packages/rocketchat-slashcommands-join/server.coffee

```diff
@@ -15,17 +15,19 @@ class Join
 		channel = channel.replace('#', '')
 
 		user = Meteor.users.findOne Meteor.userId()
-		room = RocketChat.models.Rooms.findOneByNameAndTypeNotContainingUsername(channel, 'c', user.username)
-
-		if not room?
-			RocketChat.Notifications.notifyUser Meteor.userId(), 'message', {
-				_id: Random.id()
-				rid: item.rid
-				ts: new Date
-				msg: TAPi18n.__('Channel_doesnt_exist', { postProcess: 'sprintf', sprintf: [ channel ] }, user.language)
-			}
-			return
-
+		room = RocketChat.models.Rooms.findOneByNameAndType(channel, 'c')
+		
+		RocketChat.Notifications.notifyUser Meteor.userId(), 'message', {
```

Comment:
I know, but it doesn't mean I like it :) . I'm goin to fix this now.
@engelgabriel said something about change this [solution](https://github.com/RocketChat/Rocket.Chat/pull/6268/commits/ef5795eddebfcfd5a23953a9c3bf7dfceb6d5924) to something more generic.